components/scream/src/physics/shoc/shoc_diag_second_moments_lbycond_impl.hpp

```diff
@@ -0,0 +1,62 @@
+
+#ifndef SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+#define SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+#include "physics_functions.hpp" // also for ETI not on GPUs
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::shoc_diag_second_moments_lbycond(
+    const Scalar& wthl_sfc, const Scalar& wqw_sfc, const Scalar& uw_sfc, const Scalar& vw_sfc,
+    const Scalar& ustar2, const Scalar& wstar,
+    Scalar& wthl_sec, Scalar& wqw_sec, Scalar& uw_sec, Scalar& vw_sec,
+    Scalar& wtke_sec, Scalar& thl_sec, Scalar& qw_sec, Scalar& qwthl_sec)
+{
+
+//printf("c++-1 %16.9f, %16.9f, %16.9f, %16.9f, %16.9f, %16.9f, %16.9f\n",wthl_sfc(0)[0],wqw_sfc(0)[0],ustar2(0)[0],wstar(0)[0],thl_sec(0)[0],uw_sfc(0)[0],vw_sfc(0)[0]);
+
+  // Purpose of this subroutine is to diagnose the lower
+  // boundary condition for the second order moments needed
+  // for the SHOC parameterization.
+  // The thermodymnamic, tracer, and momentum fluxes are set
+  // to the surface fluxes for the host model, while the
+  // thermodynamic variances and covariances are computed
+  // according to that of Andre et al. 1978.
+
+  const auto one_p_eight = 1.8;
```

Comment:
I'm not sure I like this approach of having every literal set to a constant. I would just use the literals inline like is done in fortran. In any case, these cannot be `const auto` because C++ will assume type double for all of these, which will break bfb in single precision. They need to be `const Scalar` if you're going to do things this way. If you're going to go back to inline literals, which I would prefer, you'll need to use the scream-precsion macro, IE `sp(1.8)`

Reply:
I also think using literals is better for clarity and less prone to unintended consequences related to `auto`. `sp(x)` isn't great, but it's not hard to understand, and I for one prefer the formulae to have numerical coefficients visible.

Reply:
change to use sp function. done!